wremnants/datasets/datagroups.py

```diff
@@ -540,28 +584,19 @@ def sum_and_unc(h):
             
         return df
 
-    def readHist(self, baseName, proc, group, syst, scaleOp=None, forceNonzero=True, scaleToNewLumi=-1):
+    def readHist(self, baseName, proc, group, syst):
         output = self.results[proc.name]["output"]
         histname = self.histName(baseName, proc.name, syst)
         logger.debug(f"Reading hist {histname} for proc/group {proc.name}/{group} and syst '{syst}'")
         if histname not in output:
             raise ValueError(f"Histogram {histname} not found for process {proc.name}")
+
         h = output[histname]
         if isinstance(h, narf.ioutils.H5PickleProxy):
+            logger.debug(f"Get narf hist")
             h = h.get()
-        # Do a copy to detach the modified object from the original one, so the latter stays unmodified.
-        # This is extremely important for fakes, since they reuse the other histograms to subtract from data
-        # If fakes are not used, or one changes the code to avoid reading everything again for fakes, one could
-        # actually modify directly the input histograms here, which might save some time compared to the copy.
-        h = h.copy()
-        if forceNonzero:
-            h = hh.clipNegativeVals(h, createNew=False)
-        if scaleToNewLumi > 0:
-            h = hh.scaleHist(h, scaleToNewLumi, createNew=False)                        
-        scale = self.processScaleFactor(proc)
-        if scaleOp:
-            scale = scale*scaleOp(proc)
-        return h*scale
```

Comment:
Is it possible to scale the histograms here or is it now assumed that they are already scaled at the histmaker output?

Reply:
The multiplication creates a new copy, it is much more performant to first project the histogram to a lower dimension and scale the smaller histogram

Reply:
Performance aside though, is it possible? Or is it now required that hists be scaled at the hitmaker?

Reply:
Yes, the histogram is scaled if the scaling factor is other than 1, but only after the projection, at line 304 in the datagroups.py